src/libraries/Native/Unix/System.Native/pal_process.c

```diff
@@ -304,7 +304,7 @@ int32_t SystemNative_ForkAndExecProcess(const char* filename,
     // Process is still the clone of this one. This is a best-effort attempt, so ignore any errors.
     // If the child fails to exec we use the pipe to pass the errno to the parent process.
 #if HAVE_PIPE2
-    pipe2(waitForChildToExecPipe, O_CLOEXEC);
+    (void)! pipe2(waitForChildToExecPipe, O_CLOEXEC);
```

Comment:
```suggestion
    (void)pipe2(waitForChildToExecPipe, O_CLOEXEC);
```

Would `(void)` be enough? It is what is used in other places.

Reply:
No, gcc expects, that result of the function must be used (assigned to variable, used in expression evaluation), not just discared: http://coliru.stacked-crooked.com/a/3571683dcc58c923

Reply:
`(void)` is sufficient to avoid "unused variable" warning, but to avoid "ignored return value of the function..." any operation (`!`) or function should be applied to functon result.